### PR TITLE
Ldaps and https support + windows fixes + sonarqube navbar fix + kibana status fix

### PR DESCRIPTION
**Pull request by mlushpenko**
# Summary

* Added options to specify ldap protocol
* Added configurable options to enable https on nginx
* Added fixes for container builds on Windows hosts
* Fixed cleanup of not used ldap variables in nginx config (was testing integration with enterprise ldap where I didn't use all vars defined in adop docker-compose and nginx was failing due to improper substitution of variables via perl) 

See: https://github.com/Accenture/adop-nginx/pull/29 
* Kibana status page is now working
* Sonarqube navbar is working with  sonarqube version 5.6.2

What I changed in docker-compose for testing:

* Used my own image
* Added NGINX_PORT, NGINX_SSL and LDAP_PROTOCOL variables

```

proxy:
  container_name: proxy
  restart: always
  image: lumaks/adop-nginx:official
  #image: accenture/adop-nginx:0.3.1
  #build: ../images/docker-nginx/
  net: ${CUSTOM_NETWORK_NAME}
  ports:
    - "80:80"
    - "443:443"
  environment:
    - "constraint:tier==public"
    - "NGINX_PORT=80"
    - "NGINX_SSL=off"
    - "LDAP_PROTOCOL=ldap"
    - "LDAP_SERVER=${LDAP_SERVER}"
    - "LDAP_USERNAME=${LDAP_ADMIN},${LDAP_FULL_DOMAIN}"
    - "LDAP_PASSWORD=${LDAP_PWD}"
    - "LDAP_USER_BASE_DN=${LDAP_USER_BASE_DN},${LDAP_FULL_DOMAIN}"
    - "LDAP_GROUP_ATTRIBUTE=member"
    - "LDAP_USER_ID_ATTRIBUTE=cn"
    - "LDAP_USER_OBJECT_CLASS=inetOrgPerson"
```

All images are running:

```
0c7ed2f91891        lumaks/adop-nginx:official           "/resources/script..."   2 minutes ago       Up 2 minutes                   0.0.0.0:80->80/tcp, 0.0.0.0:443->443/tcp                                     proxy
ce7e0b42da19        accenture/adop-jenkins:0.2.6         "/entrypoint.sh"         38 minutes ago      Up 38 minutes                  8080/tcp, 0.0.0.0:50000->50000/tcp                                           jenkins
c487c1aed7a4        mysql:5.6.25                         "/entrypoint.sh my..."   38 minutes ago      Up 38 minutes                  3306/tcp                                                                     gerrit-mysql
40f5d992dbb9        selenium/hub:2.53.0                  "/opt/bin/entry_po..."   38 minutes ago      Up 38 minutes                  4444/tcp                                                                     selenium-hub
2a3eac37ed62        rabbitmq:3.5.7-management            "/docker-entrypoin..."   38 minutes ago      Up 38 minutes                  4369/tcp, 5671-5672/tcp, 15671-15672/tcp, 25672/tcp                          sensu-rabbitmq
12726c692458        sstarcher/uchiwa:0.15.0              "/bin/start"             38 minutes ago      Up 38 minutes                  3000/tcp                                                                     sensu-uchiwa
daa08a63dbed        accenture/adop-sensu:0.2.0           "/bin/start server"      38 minutes ago      Up 38 minutes                  4567/tcp                                                                     sensu-server
66e4b3e925eb        accenture/adop-ldap-ltb:0.1.0        "/etc/service/apac..."   38 minutes ago      Up 38 minutes                  80/tcp                                                                       ldap-ltb
8bb7d38407dd        redis:3.0.7                          "docker-entrypoint..."   38 minutes ago      Up 38 minutes                  6379/tcp                                                                     sensu-redis
adfa10b2cc4f        accenture/adop-ldap-phpadmin:0.1.0   "/run.sh"                38 minutes ago      Up 38 minutes                  80/tcp                                                                       ldap-phpadmin
5857bf050c08        registry:2.5.1                       "/entrypoint.sh /e..."   38 minutes ago      Restarting (1) 4 seconds ago                                                                                registry
576174ae1ec8        accenture/adop-sensu:0.2.0           "/bin/start api"         38 minutes ago      Up 38 minutes                  4567/tcp                                                                     sensu-api
b4edc46c0c2c        selenium/node-firefox:2.53.0         "/opt/bin/entry_po..."   38 minutes ago      Up 38 minutes                                                                                               selenium-node-firefox
353e9b8c408c        accenture/adop-jenkins-slave:0.1.4   "/bin/sh -c 'java ..."   38 minutes ago      Up 38 minutes                                                                                               jenkins-slave
bf8b74a60fc4        selenium/node-chrome:2.53.0          "/opt/bin/entry_po..."   38 minutes ago      Up 38 minutes                                                                                               selenium-node-chrome
1b82bb1c678a        mysql:5.6.25                         "/entrypoint.sh my..."   38 minutes ago      Up 38 minutes                  3306/tcp                                                                     sonar-mysql
f335db356c6b        accenture/adop-ldap:0.1.3            "/usr/local/bin/en..."   38 minutes ago      Up 38 minutes                  0.0.0.0:389->389/tcp                                                         ldap
ae8ae3538486        accenture/adop-nexus:0.1.3           "/usr/local/bin/ne..."   38 minutes ago      Up 38 minutes                  8081/tcp                                                                     nexus
a33ee7f8dc34        accenture/adop-sensu:0.2.0           "/bin/start client"      38 minutes ago      Up 38 minutes                  4567/tcp                                                                     sensu-client
181f22b6709d        accenture/adop-gerrit:0.1.3          "/var/gerrit/gerri..."   38 minutes ago      Up 37 minutes                  8080/tcp, 29418/tcp                                                          gerrit
bcb0f7db760d        accenture/adop-sonar:0.3.2           "/usr/local/bin/so..."   38 minutes ago      Up 38 minutes                  9000/tcp                                                                     sonar
4769a8476440        accenture/adop-logstash:0.1.0        "/docker-entrypoin..."   38 minutes ago      Up 38 minutes                  0.0.0.0:12201->12201/udp, 0.0.0.0:5000->5000/tcp, 0.0.0.0:25826->25826/udp   logstash
8677649cb4c7        elasticsearch:2.1.1                  "/docker-entrypoin..."   38 minutes ago      Up 38 minutes                  0.0.0.0:9200->9200/tcp, 9300/tcp                                             elasticsearch
a06e9d4f425c        kibana:4.3.1                         "/docker-entrypoin..."   38 minutes ago      Up 38 minutes                  0.0.0.0:5601->5601/tcp                                                       kibana
```

Platform is up
image

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+language: bash
+
+sudo: required
+services:
+  - docker
+
+script:
+  - docker build -t adop-nginx .
```
Comment (mlushpenko) [inline .travis.yml]: This file was changed before my additions (squashed one commit too far)

File: README.md [hunk 2]
```diff
@@ -1,7 +1,11 @@
-#Supported tags and respective Dockerfile links
+# Supported tags and respective Dockerfile links
 
 - [`0.1.0`, `0.1.0` (*0.1.0/Dockerfile*)](https://github.com/Accenture/adop-nginx/blob/master/Dockerfile.md)
 
+# Build Status
+
+[![Build Status](https://travis-ci.org/Accenture/adop-nginx.svg?branch=master)](https://travis-ci.org/Accenture/adop-nginx)
+
```
Comment (mlushpenko) [inline README.md]: This change was done before my additions (squashed one commit too far)

File: README.md [hunk 3]
```diff
@@ -1,7 +1,11 @@
-#Supported tags and respective Dockerfile links
+# Supported tags and respective Dockerfile links
```
Comment (mlushpenko) [inline README.md]: This change was done before my additions (squashed one commit too far)

File: resources/configuration/sites-enabled/service-extension/service-extension.conf.template [hunk 4]
```diff
@@ -1,3 +1,3 @@
 location /<service-name>{
-   proxy_pass  http://<service-container>:<port>
+   proxy_pass  http://<service-container>:<port>;
```
Comment (mlushpenko) [inline resources/configuration/sites-enabled/service-extension/service-extension.conf.template]: This change was done before my additions (squashed one commit too far)

**Comment by svscorp** [comment]
@nickdgriffin @SachinKSingh28 @anton-kasperovich 

Hi team, please have a look. We have validated it on-prem.

**Comment by svscorp** [comment]
+1

**Comment by svscorp** [comment]
I wonder, why there is a conflict, @mlushpenko ?

**Comment by mlushpenko** [comment]
@svscorp I made tools-context.conf a template to be able to set nginx params, so my version must be accepted. The process is exactly the same as with nginx.conf.

**Comment by svscorp** [comment]
@mlushpenko @nickdgriffin 

Okay, after testing this extensively, few issues found. Enabling SSL causes jenkins misbehavior (with the "reverse proxy is broken" + redirect to http when saving job, going to PluginManager, etc...

Another issue is that it is using one config file where you listen either for 80 or 443, I would do it with two separate configs, **tool-context** and **tool-context-ssl** something like that. Your opinion, @nickdgriffin ?

Another issue I'd say - this PR is too big. It contains: 
1. [stable] refactoring (spaces, semicolons, html folder creation in dockerfile)
2. [stable] fix for sonar
3. [stable] kibana status page
4. [stable] ldaps
5. **[non-stable]** ssl

Therefore, bigger chances to merge is to split it. 
For point number **2** and **3** let's just merge PR #29 it's tested multiple times already.

**Comment by mlushpenko** [comment]
@svscorp @nickdgriffin one more thing I just found which is probably related to ssl configuration that if I enter just http link, I get authenticatio popup and after providing credentials selenuim grid opens up as a main page. 
image
While when I go to htttps, I get proper adop page
image

**Comment by nickdgriffin** [comment]
@svscorp I'd rather have a single configuration file that didn't mean changes had to be made in two places every time one of the locations changed, as it will just mean one copy isn't maintained. Alternatively the config can be split up so that the locations are in separate files that are imported, then the SSL/no SSL part can also be a separate file.

@mlushpenko I expect that's because of how that domain is configured, it's probably on HTTP as well as so "catches" it.

File: templates/configuration/sites-enabled/tools-context.conf [hunk 5]
```diff
@@ -64,11 +64,15 @@ server {
...
         proxy_pass  http://gerrit:8080;
+        proxy_set_header  X-Forwarded-For $remote_addr;
+        proxy_set_header  Host $host;
     }
 
     location /jenkins {
         proxy_pass http://jenkins:8080;
         proxy_set_header Host $host;
+        proxy_read_timeout  90;
+        proxy_redirect      http:// ###REDIRECT_PROTOCOL###://;
```
Comment (svscorp) [inline templates/configuration/sites-enabled/tools-context.conf]: Hi @mlushpenko, just curious is it okay that there is a space in the proxy_redirect? :) between "http://" and "###REDIRECT_..."

Comment (mlushpenko) [reply templates/configuration/sites-enabled/tools-context.conf]: @svscorp yes, it is. We redirect http:// to https:// (or to http, which doesn't do anything but provides possiblity to configure jenkins for http or https) Space is needed

Comment (mlushpenko) [reply templates/configuration/sites-enabled/tools-context.conf]: I tested both setups, both are working well, but please review when you have time as well

**Comment by dsingh07** [comment]
Hi @mlushpenko @svscorp 

Thanks for the PR. There are a lot of changes that this PR brings that aren't quite atomic and related to each other.

Is there any chance we can split this PR into multiple smaller PRs please? It just means that we can capture one standalone change in each PR and also it makes it easier to test the changes atomically.

For example the change to fix the Sonarqube nav bar is now needed for the new release of the Sonar image we have but we don't want to create a dependency on testing all the other changes that this PR brings in.

/CC @nickdgriffin @SachinKSingh28

**Comment by svscorp** [comment]
I am closing this PR with favor of #42 - refactored, rebased, squashed.

**Pull request closed without merging**